lib/local-server/index.js

```diff
@@ -16,46 +16,73 @@ module.exports = class Server {
     this.results = [] // results of all tests run
     this.passedFocus = [] // all tests that passed after running
 
+    const {app, server} = this.createWebApp(Zen.config)
+    this.startWebpackServer(Zen.webpack, app)
+
+    Zen.s3Sync.on('status', this.sendStatus.bind(this))
+    new WebSocket.Server({server}).on('connection', this.onWebsocket.bind(this))
+
+    this.chrome = new ChromeWrapper() // headless chrome instance
+    this.chrome.launchLocal({port: 9222})
+
+    this.startWorkers(Zen.config, app)
+  }
+
+  createWebApp(config) {
     // start up the local webserver for `head` to connect to
-    let app = connect()
-    let server = http.createServer(app).listen(Zen.config.port)
+    const app = connect()
+    const server = http.createServer(app).listen(config.port)
     app.use('/lib', Util.serveWith404(path.join(__dirname, '..'))) // serve up stuff out of lib
-    app.use('/node_modules', Util.serveWith404(path.resolve(Zen.config.appRoot, './node_modules'))) // TODO: this doesn't work when yarn link'd
-    app.use('/base', Util.serveWith404(Zen.config.appRoot)) // base serves things out of the application's root
+    app.use('/node_modules', Util.serveWith404(path.resolve(config.appRoot, './node_modules'))) // TODO: this doesn't work when yarn link'd
+    app.use('/base', Util.serveWith404(config.appRoot)) // base serves things out of the application's root
     app.use('/svelte', Util.serveSvelte)
     app.use('/icons', Util.serveIcons)
 
-    if (Zen.webpack) {
-      Zen.webpack.startDevServer(app)
-      Zen.webpack.on('status', stats => {
-        if (Zen.webpack.status == 'done') {
-          this.workers.forEach(w => w.tab && w.tab.setCodeHash(Zen.webpack.compile.hash))
+    // host worker and head
+    app.use(async (req, resp, next) => {
+      let pageType
+      if (req.url.match(/^\/worker/)) {
+        pageType = 'worker'
+      } else {
+        const pathname = req._parsedUrl.pathname
+        if (pathname.match(/^\/(index(\.htm(l)?)?)?$/)) {
+          pageType = 'head'
+        }
+      }
+
+      if (pageType) {
+        resp.end(Zen.indexHtml(pageType))
+      } else {
+        next()
+      }
+    })
```

Comment:
This seems much bigger than the 1 line it was previously.

Reply:
As far as I can tell the previous 1 line was sending `head.js` to any request that failed to match a route. This new code should send a 404 if an unknown asset is requested.